app/validators/target_validator.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class TargetValidator < ActiveModel::Validator
+  def validate(record)
+    return if record.user.targets_count < Target::MAX_USER_TARGETS
```

Comment:
I was actually thinking about moving this logic to the service itself insteaf of using any form of model validations. But this also works, it up to you 🙂